PolyEngine/Core/Src/Vector2i.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "Defines.hpp"
+#include "BasicMath.hpp"
+
+namespace Poly
+{
+
+	using VectorIntType = int32_t;
+
+	/// <summary>Class representing 2D integer vector.</summary>
+	class ALIGN_16 CORE_DLLEXPORT Vector2i : public BaseObject<>{
+	public:
+		static const Vector2i ZERO;
+		static const Vector2i UNIT_X;
+		static const Vector2i UNIT_Y;
+
+		/// <summary>Creates zero vector.</summary>
+		constexpr Vector2i() : X(0), Y(0) {}
+
+		/// <summary>Creates vector from int values.</summary>
+		/// <param name="x"></param>
+		/// <param name="y"></param>
+		constexpr Vector2i(VectorIntType x, VectorIntType y) : X(x), Y(y) {}
+
+
+		inline Vector2i(const Vector2i& rhs) : X(rhs.X), Y(rhs.Y) {}
+		inline Vector2i& operator=(const Vector2i& rhs) { X = rhs.X; Y = rhs.Y; return *this; }
+
+		/// <summary>Creates Vector2f based on this vector.</summary>
+		/// <returns>Converted vector.</returns>
+		class Vector2f ToVector2f() const;
```

Comment:
Please move class declaration to the begining of the namespace (codding style). Please provide explicit cast operator to Vector2f.